influx-writer/influx-writer.py

```diff
@@ -15,29 +14,78 @@
 from time import time, sleep
 import config
 from listenstore import InfluxListenStore
+from requests.exceptions import ConnectionError
+from redis import Redis
+from redis_keys import INCOMING_QUEUE_SIZE_KEY, UNIQUE_QUEUE_SIZE_KEY
 
 REPORT_FREQUENCY = 5000
-SUBSCRIBER_NAME = "in"
-KEYSPACE_NAME_INCOMING = "ilisten"
-KEYSPACE_NAME_UNIQUE = "ulisten"
-
-
 DUMP_JSON_WITH_ERRORS = False
+ERROR_RETRY_DELAY = 3 # number of seconds to wait until retrying an operation
+
 
-class InfluxWriterSubscriber(RedisPubSubSubscriber):
-    def __init__(self, ls, influx, redis):
-        RedisPubSubSubscriber.__init__(self, redis, KEYSPACE_NAME_INCOMING, __name__)
+class InfluxWriterSubscriber(object):
+    def __init__(self):
+        self.log = logging.getLogger(__name__)
+        logging.basicConfig()
+        self.log.setLevel(logging.INFO)
 
-        self.publisher = RedisPubSubPublisher(redis, KEYSPACE_NAME_UNIQUE)
+        self.ls = None
+        self.influx = None
+        self.redis = None
 
-        self.influx = influx
-        self.ls = ls
+        self.incoming_ch = None
+        self.unique_ch = None
+        self.connection = None
         self.total_inserts = 0
         self.inserts = 0
         self.time = 0
 
-    def write(self, listen_dicts):
 
+    @staticmethod
+    def static_callback(ch, method, properties, body, obj):
+        return obj.callback(ch, method, properties, body)
+
+
+    def connect_to_rabbitmq(self):
+        while True:
+            try:
+                self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=config.RABBITMQ_HOST, port=config.RABBITMQ_PORT))
+                break
+            except Exception as e:
+                self.log.error("Cannot connect to rabbitmq: %s, sleeping 2 seconds")
+                sleep(2)
+
+
+    def callback(self, ch, method, properties, body):
+        listens = ujson.loads(body)
+        ret = self.write(listens)
+        if not ret:
+            return ret
+
+        while True:
+            try:
+                self.incoming_ch.basic_ack(delivery_tag = method.delivery_tag)
+                break
+            except pika.exceptions.ConnectionClosed:
+                self.connect_to_rabbitmq()
+            
+        count = len(listens)
+        self.redis.decr(INCOMING_QUEUE_SIZE_KEY, count)
+
+        # collect and occasionally print some stats
+        self.inserts += count
+        if self.inserts >= REPORT_FREQUENCY:
+            self.total_inserts += self.inserts
+            if self.time > 0:
+                self.log.info("Inserted %d rows in %.1fs (%.2f listens/sec). Total %d rows." % \
+                    (self.inserts, self.time, self.inserts / self.time, self.total_inserts))
+            self.inserts = 0
+            self.time = 0
+
+        return ret
+
+
+    def write(self, listen_dicts):
```

Comment:
That's my point though, it's kind of difficult to read.

Example of one of the changes that can be made:
```python
@staticmethod
def listens_time_range(listens):
    """Calculate the time range that a set of listens covers.
    
    Args:
        listens (list): List of dictionaries with listen data.
    
    Returns:
        Tuple with two values: minimum time and maximum time.
    """
    min_time = 0
    max_time = 0
    for listen in listens:
        t = int(listen['listened_at'])
        if not max_time:
            min_time = max_time = t
            continue
        if t > max_time:
            max_time = t
        if t < min_time:
            min_time = t
    return min_time, max_time

def write(self, listen_dicts):
    submit = []
    unique = []

    min_time, max_time = listens_time_range(listen_dicts)

    ...
```
I don't need to know how `listens_time_range` is implemented to understand how `write` works. `listens_time_range` is also very easy to test now.